kv/migration.go

```diff
@@ -0,0 +1,332 @@
+package kv
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"time"
+
+	influxdb "github.com/influxdata/influxdb"
+	"go.uber.org/zap"
+)
+
+var (
+	migrationBucket = []byte("migrationsv1")
+
+	// ErrMigrationSpecNotFound is returned when a migration specification is missing
+	// for an already applied migration.
+	ErrMigrationSpecNotFound = errors.New("migration specification not found")
```

Comment:
be good to make this an `inflxudb.Error` type with the necessary code/msg

Reply:
This kind of error shouldn't really ever reach a user

Reply:
It happens on initialize or through manual invocation via a cli.